examples/counter-ts/core/reducers/counter.ts

```diff
@@ -0,0 +1,21 @@
+import { CounterActionType } from "../constants";
+
+export interface CounterState {
+  value: number;
+}
+
+const INITIAL_STATE = {
+  value: 0,
+};
+
+export default function counter(state: CounterState = INITIAL_STATE, action: { type: CounterActionType }): CounterState {
+  switch (action.type) {
+    case CounterActionType.INCREMENT_COUNTER:
+      state.value += 1;
```

Comment:
I thought we were working with the Redux mentality, state should not be mutated.